astropy/wcs/wcs.py

```diff
@@ -388,8 +396,8 @@ def __init__(self, header=None, fobj=None, key=' ', minerr=0.0,
                         "header must be a string, an astropy.io.fits.Header "
                         "object, or a dict-like object")
 
-            header_string = header_string.strip()
-
```

Comment:
I think that line may just have been removed in error -- there was lots of juggling how the header is parsed here so we can edit it and pass the edited version (with SIP removed for example) to wcslib.  

But I'm not sure how the strip would help here -- it wouldn't remove any blank FITS lines unless they were at the end.